web/api/v1/api.go

```diff
@@ -1194,6 +1194,13 @@ func (api *API) metricMetadata(r *http.Request) apiFuncResult {
 			return apiFuncResult{nil, &apiError{errorBadData, errors.New("limit must be a number")}, nil, nil}
 		}
 	}
+	limitPerMetric := math.MaxInt
+	if s := r.FormValue("limit-per-metric"); s != "" {
+		var err error
+		if limitPerMetric, err = strconv.Atoi(s); err != nil {
```

Comment:
What if we get a negative number here?

Reply:
Good point, I feel like there should be a validation for negative numbers, but since the former param `limit` also doesn't have this validation I think we should keep them consistent

Maybe we could add the validation in another PR?

Reply:
The distinction is that `limit` is allowed to be 0. Below usages check `if limit >= 0` and use `-1` to represent "no limit". Maybe to keep things consistent, we should follow the same convention instead of validation?

Reply:
OK for -1 being unlimited

Reply:
Nice, I think to address this @ArthurSens we need to keep consistency with `limit` variable, so either (A) adding MaxInt idea to `limit` or (B) (preferably, looking by comments) bringing `limit` flow to `limitPerMetric`

For (B) it would be something like `limitPerMetric := -1` and each check `if limitPerMetric > 0 && len(ms) >= limitPerMetric {` 

The thing is that MaxInt vs -1 flow is, in this function, kind of equally complex (just in different places). However given limit already uses something, consistency would give us nice readability gains